File: sdk/java-sdk/src/main/java/com/evernym/verity/sdk/protocols/updateendpoint/common/Authentication.java
package com.evernym.verity.sdk.protocols.updateendpoint.common;

import com.evernym.verity.sdk.utils.AsJsonObject;
import org.json.JSONObject;

public class Authentication implements AsJsonObject {
    final JSONObject json;

    public Authentication(String type, String version, JSONObject data) {
        json = new JSONObject();
        json.put("type", type);
        json.put("version", version);
        json.put("data", data);
    }

    @Override
    public JSONObject toJson() {
        return json;
    }
}

File: sdk/dotnet-sdk/src/VeritySDK/Protocols/UpdateEndpoint/Common/Authentication.cs
using System.Json;
using VeritySDK.Utils;

namespace VeritySDK.Protocols.UpdateEndpoint
{
    /// <summary>
    /// A holder for a Authentication expression
    /// </summary>
    public class Authentication : AsJsonObject
    {
        JsonObject json;

        /// <summary>
        /// Constructor
        /// </summary>
        /// <param name="type">Type of authentication</param>
        /// <param name="version">Version of authentication implementation</param>
        /// <param name="data">Additional data</param>
        public Authentication(string type, string version, JsonObject data)
        {
            this.json = new JsonObject();
            this.json.Add("type", type);
            this.json.Add("version", version);
            this.json.Add("data", data);
        }

        /// <summary>
        /// Convert this object to a JSON object
        /// </summary>
        /// <returns>this object as a JSON object</returns>
        public JsonObject toJson()
        {
            return this.json;
        }
    }
}
